### ISSUE TITLE
build for iOS - KeyError: 'DEPLOYMENT_TARGET_CLANG_FLAG_NAME'

### ISSUE THREAD

**Issue by fpattis** [NativeScript/NativeScript#9840]
### Issue Description

when executing `ns debug iOS` I get the error:

```
Xcode build...
2022-03-23 13:04:13.549 xcodebuild[11615:265896] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionSentinelHostApplications for extension Xcode.DebuggerFoundation.AppExtensionHosts.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
2022-03-23 13:04:13.550 xcodebuild[11615:265896] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionPointIdentifierToBundleIdentifier for extension Xcode.DebuggerFoundation.AppExtensionToBundleIdentifierMap.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
--- xcodebuild: WARNING: Using the first of multiple matching destinations:
{ platform:iOS Simulator, id:dvtdevice-DVTiOSDeviceSimulatorPlaceholder-iphonesimulator:placeholder, name:Any iOS Simulator Device }
{ platform:iOS Simulator, id:CC73F5A2-755E-4D19-901A-45210A8EE541, OS:15.4, name:iPad (9th generation) }
{ platform:iOS Simulator, id:66872C89-DA20-4FE8-84C0-D0943AE4CABE, OS:15.4, name:iPad Air (5th generation) }
{ platform:iOS Simulator, id:CC3CA3B8-F0AF-4C63-8DCE-5C0D353C94EC, OS:15.4, name:iPad Pro (9.7-inch) }
{ platform:iOS Simulator, id:96CDF0D2-76D3-4978-BC65-5A6D2A45E5DC, OS:15.4, name:iPad Pro (11-inch) (3rd generation) }
{ platform:iOS Simulator, id:FAEF9EE7-F2AD-4218-B5FD-FAF74E357297, OS:15.4, name:iPad Pro (12.9-inch) (5th generation) }
{ platform:iOS Simulator, id:3C539F7C-B436-4A8E-AC25-7026EF31CC7D, OS:15.4, name:iPad mini (6th generation) }
{ platform:iOS Simulator, id:A4A7D59A-0489-44DC-A8C2-F2B79E21E669, OS:15.4, name:iPhone 8 }
{ platform:iOS Simulator, id:EE10BAA4-A107-4D0B-910C-9CE738F8DAD7, OS:15.4, name:iPhone 8 Plus }
{ platform:iOS Simulator, id:353A1203-89B9-46A1-95D6-CB3402D728B9, OS:15.4, name:iPhone 11 }
{ platform:iOS Simulator, id:FC4ECBA1-479F-4CF7-82A9-E7811C1ABA7C, OS:15.4, name:iPhone 11 Pro }
{ platform:iOS Simulator, id:C264CD6C-683F-43B3-A5DB-5C0C1ED48367, OS:15.4, name:iPhone 11 Pro Max }
{ platform:iOS Simulator, id:944F6BD7-1CE4-455E-B357-F4F870F51C7C, OS:15.4, name:iPhone 12 }
{ platform:iOS Simulator, id:A64B5A08-5A05-4733-9BBC-6E7DEDC1F476, OS:15.4, name:iPhone 12 Pro }
{ platform:iOS Simulator, id:91872F8C-9B83-478A-8A15-0618AB5FD71F, OS:15.4, name:iPhone 12 Pro Max }
{ platform:iOS Simulator, id:C96BE06F-1602-4C52-A27C-6806F0F531BA, OS:15.4, name:iPhone 12 mini }
{ platform:iOS Simulator, id:2E10A4F6-3C0D-411F-BDD9-2A2A9C51FEEE, OS:15.4, name:iPhone 13 }
{ platform:iOS Simulator, id:7E0DAAED-B01C-4406-A457-DC811F2267F8, OS:15.4, name:iPhone 13 Pro }
{ platform:iOS Simulator, id:0C5A8BF0-237B-4F43-98DE-80509C07D18A, OS:15.4, name:iPhone 13 Pro Max }
{ platform:iOS Simulator, id:62D24046-C1AA-4DF1-A807-94E62FDE661F, OS:15.4, name:iPhone 13 mini }
{ platform:iOS Simulator, id:43EDBEF7-2834-441B-ACC2-BF1667C88897, OS:15.4, name:iPhone SE (3rd generation) }
{ platform:iOS Simulator, id:B2399736-8B07-4DA3-B7F9-B0B52EC3A674, OS:15.4, name:iPod touch (7th generation) }
{ platform:macOS, arch:x86_64, variant:Mac Catalyst, id:89173451-667B-58DD-9001-0CBDBB0668F5 }
{ platform:iOS, id:dvtdevice-DVTiPhonePlaceholder-iphoneos:placeholder, name:Any iOS Device }
{ platform:macOS, variant:Mac Catalyst, name:Any Mac }
./.build_env_vars.sh: line 468: declare: UID: readonly variable
NSLD: Swift bridging header '*-Swift.h' not found under '/Library/Developer/Xcode/DerivedData/framework2app-brkncvafjpmmvraiejihjffyplkq/Build/Intermediates.noindex/framework2app.build/Debug-iphonesimulator/framework2app.build/Objects-normal/x86_64'
Generating metadata...~/projects/framework2-app/platforms/ios/internal/metadata-generator/bin ~/projects/framework2-app/platforms/ios
Python version: 2.7.18 (default, Jan  4 2022, 17:47:56) 
[GCC Apple LLVM 13.0.0 (clang-1300.0.29.10) [+internal-os, ptrauth-isa=deployme
Traceback (most recent call last):
  File "./build-step-metadata-generator.py", line 41, in <module>
    deployment_target_flag_name = env("DEPLOYMENT_TARGET_CLANG_FLAG_NAME")
  File "./build-step-metadata-generator.py", line 14, in env
    return os.environ[env_name]
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/UserDict.py", line 40, in __getitem__
    raise KeyError(key)
KeyError: 'DEPLOYMENT_TARGET_CLANG_FLAG_NAME'
Command Ld failed with a nonzero exit code
Command Ld failed with a nonzero exit code
note: Using new build system
note: Using codesigning identity override: 
note: Planning
note: Build preparation complete
note: Building targets in dependency order
```

I get the same error when I try to build from xcode directly selecting a specific emulator. xcode furthermore tells

```

Showing Recent Messages
platforms/ios/framework2app.xcodeproj Building for iOS Simulator, but the linked and embedded framework 'TNSAutoComplete.framework' was built for iOS + iOS Simulator.

platforms/ios/framework2app.xcodeproj Building for iOS Simulator, but the linked and embedded framework 'AppSync.framework' was built for iOS + iOS Simulator.

platforms/ios/framework2app.xcodeproj Building for iOS Simulator, but the linked and embedded framework 'TNSChart.framework' was built for iOS + iOS Simulator.

platforms/ios/framework2app.xcodeproj Building for iOS Simulator, but the linked and embedded framework 'BarcodeScannerFramework.framework' was built for iOS + iOS Simulator.

platforms/ios/framework2app.xcodeproj Building for iOS Simulator, but the linked and embedded framework 'TNSListView.framework' was built for iOS + iOS Simulator.

platforms/ios/framework2app.xcodeproj Building for iOS Simulator, but the linked and embedded framework 'TNSSideDrawer.framework' was built for iOS + iOS Simulator.

platforms/ios/framework2app.xcodeproj Building for iOS Simulator, but the linked and embedded framework 'TNSCore.framework' was built for iOS + iOS Simulator.


```

I found similar errors related to flutter but none of the solutions worked:

- https://stackoverflow.com/questions/69306519/flutter-error-xcodebuild-warning-using-the-first-of-multiple-matching-destina

thank you

### Reproduction

ns doctor
✔ Getting environment information 

No issues were detected.
✔ Your ANDROID_HOME environment variable is set and points to correct directory.
✔ Your adb from the Android SDK is correctly installed.
✔ The Android SDK is installed.
✔ A compatible Android SDK for compilation is found.
✔ Javac is installed and is configured properly.
✔ The Java Development Kit (JDK) is installed and is configured properly.
✔ Xcode is installed and is configured properly.
✔ xcodeproj is installed and is configured properly.
✔ CocoaPods are installed.
✔ CocoaPods update is not required.
✔ CocoaPods are configured properly.
✔ Your current CocoaPods version is newer than 1.0.0.
✔ Python installed and configured correctly.
✔ The Python 'six' package is found.
✔ Xcode version 13.3.0 satisfies minimum required version 10.
✔ Getting NativeScript components versions information...
⚠ Update available for component nativescript. Your current version is 7.2.1 and the latest available version is 8.2.2.
⚠ Update available for component @nativescript/core. Your current version is 7.3.0 and the latest available version is 8.2.1.
⚠ Update available for component @nativescript/ios. Your current version is 7.2.0 and the latest available version is 8.2.2.
⚠ Update available for component @nativescript/android. Your current version is 8.1.1 and the latest available version is 8.2.2.

### Environment

```yaml
OS: macOS 12.2.1
CPU: (16) x64 Intel(R) Core(TM) i7-10700K CPU @ 3.80GHz
Shell: /bin/zsh
node: 14.18.1
npm: 6.14.15
nativescript: 7.2.1

# android
java: 15.0.2
ndk: Not Found
apis: Not Found
build_tools: Not Found
system_images: Not Found

# ios
xcode: 13.3/13E113
cocoapods: 1.11.2
python: 2.7.18
python3: 3.8.9
ruby: 2.7.4
platforms: 
  - DriverKit 21.4
  - iOS 15.4
  - macOS 12.3
  - tvOS 15.4
  - watchOS 8.5
```

### Dependencies

```json
"dependencies": {
  "@mdi/font": "^5.9.55",
  "@nativescript-community/ui-lottie": "^1.0.0",
  "@nativescript-community/ui-mapbox": "^6.2.8",
  "@nativescript/appsync": "^3.0.1",
  "@nativescript/appversion": "^2.0.0",
  "@nativescript/background-http": "^5.0.2",
  "@nativescript/camera": "^5.0.7",
  "@nativescript/core": "^7.3.0",
  "@nativescript/fingerprint-auth": "^7.0.0",
  "@nativescript/firebase": "^11.1.3",
  "@nativescript/geolocation": "^8.0.1",
  "@nativescript/imagepicker": "^1.0.3",
  "@nativescript/iqkeyboardmanager": "^2.0.0",
  "@nativescript/secure-storage": "^3.0.0",
  "@nativescript/theme": "^3.0.0",
  "@nativescript/unit-test-runner": "^1.0.2",
  "@proplugins/nativescript-drop-down": "^9.0.2",
  "@proplugins/nativescript-zxing": "^1.7.2",
  "@triniwiz/nativescript-toasty": "^4.1.3",
  "axios": "^0.21.1",
  "base-64": "^1.0.0",
  "intl": "^1.2.5",
  "knex": "^0.21.19",
  "lodash": "^4.17.21",
  "md5": "^2.3.0",
  "mime": "^2.5.2",
  "moment": "^2.29.1",
  "nativescript-barcodescanner": "^4.0.1",
  "nativescript-calendar": "^3.0.0",
  "nativescript-download-progress": "1.4.3",
  "nativescript-exit": "^1.0.1",
  "nativescript-feedback": "^2.0.0",
  "nativescript-menu": "^1.1.7",
  "nativescript-nfc": "^5.0.0",
  "nativescript-numeric-keyboard": "^5.0.0",
  "nativescript-pdf-view": "^3.0.0-1",
  "nativescript-permissions": "^1.3.11",
  "nativescript-sqlite": "^2.8.6",
  "nativescript-ui-autocomplete": "^7.0.2",
  "nativescript-ui-chart": "^8.0.2",
  "nativescript-ui-listview": "^9.1.0",
  "nativescript-ui-sidedrawer": "^9.0.3",
  "nativescript-videoplayer": "^5.0.1",
  "nativescript-vue": "^2.9.0",
  "nativescript-webview-interface": "^1.4.4",
  "nativescript-windowed-modal": "^7.0.0",
  "node-html-parser": "^2.2.1",
  "validator": "^13.5.2",
  "vue-async-computed": "^3.9.0",
  "vue-i18n": "^8.24.2",
  "vuex": "^3.6.2"
},
"devDependencies": {
  "@babel/core": "^7.13.14",
  "@babel/preset-env": "^7.13.12",
  "@nativescript/android": "8.1.1",
  "@nativescript/eslint-plugin": "^0.0.4",
  "@nativescript/ios": "7.2.0",
  "@nativescript/webpack": "^4.1.0",
  "@typescript-eslint/parser": "^4.20.0",
  "babel-eslint": "^10.1.0",
  "babel-loader": "^8.2.2",
  "chai": "^4.3.4",
  "del-cli": "^3.0.1",
  "eslint": "^7.23.0",
  "eslint-plugin-mocha": "^8.1.0",
  "eslint-plugin-standard": "^5.0.0",
  "eslint-plugin-vue": "^7.8.0",
  "karma": "^6.3.2",
  "karma-chai": "^0.1.0",
  "karma-mocha": "^2.0.1",
  "karma-nativescript-launcher": "^0.4.0",
  "karma-webpack": "^5.0.0",
  "mocha": "^8.3.2",
  "nativescript-vue-template-compiler": "^2.9.0",
  "sass": "^1.43.4",
  "stylelint": "^13.12.0",
  "stylelint-config-standard": "^20.0.0",
  "stylelint-scss": "^3.18.0",
  "vue-loader": "^15.9.6"
}
```

### Please accept these terms

- [X] I have searched the [existing issues](https://github.com/NativeScript/NativeScript/issues) as well as [StackOverflow](https://stackoverflow.com/questions/tagged/nativescript) and this has not been posted before
- [X] This is a bug report
- [X] I agree to follow this project's [Code of Conduct](https://github.com/NativeScript/NativeScript/blob/master/tools/notes/CONTRIBUTING.md#coc)

**Comment by richardman1** [comment]
Watching this regarding NS6

**Comment by rigor789** [comment]
If anyone wants to open a PR to the 6.5 runtime based on the changes here: https://github.com/NativeScript/ns-v8ios-runtime/commit/1e0c0cec0e9627cd72652208347e760809f7d1e1 we can publish a patch.

https://github.com/NativeScript/ios-runtime

**Comment by fpattis** [comment]
thank youi, I created a pull request: https://github.com/NativeScript/ios-runtime/pull/1296

**Comment by a92c** [comment]
what needs to be done to solve?

**Comment by rigor789** [comment]
@a92c if you are on nativescript 7/8.x you can just install `@nativescript/ios@8.2.2`

If you are still using 6.5.x runtime, you'll need to wait until we merge @fpattis's PR and do a release with it - will be 6.5.5 likely...

**Comment by dlcole** [comment]
@rigor789 

> if you are on nativescript 7/8.x you can just install @nativescript/ios@8.2.2

I upgraded to {N} 8.2.2 but ran into build errors with @nativescript/firebase.  Migrating to @nativescript/firebase-core will be a large effort as the API is different.  I fell back to {N} 8.1.5 and tried `tns platform add ios@8.2.2` but version 8.1.0 was installed, and now I'm back at this issue.  

Is it possible to run @nativescript/ios@8.2.2 with {N} 8.1.5?  I need to publish my app soon and I'm worried about (a currently being unable to build on either Android or iOS and b) just how much work may be involved.

**Comment by rigor789** [comment]
@dlcole you should be able to set `@nativescript/ios` version in package.json to 8.2.2 and then `ns clean` and build again.

**Comment by dlcole** [comment]
@rigor789 that works!  Thanks so much!

**Comment by richardman1** [comment]
When is this getting released in ios 6.5.5 version? I really need it

**Comment by rigor789** [comment]
Released https://github.com/NativeScript/ios-runtime/releases/tag/v6.5.5
